behaviours/meme_rater.py

```diff
@@ -62,68 +62,95 @@ def number_emoji(number: int):
     return hikari.UnicodeEmoji.parse(unicode)
 
 
-async def msg_update(event: hikari.GuildMessageUpdateEvent) -> None:
-    if event.channel_id != MEME_CHANNEL_ID or not event.message.embeds:
-        return
-
-    # We only want to rate the meme if it was not edited after the initial post
-    # Discord does not add an edited timestamp when embedding an image
-    if event.message.edited_timestamp:
-        return
+async def _process_attachment(
+    attachment: hikari.Attachment, username: str | None
+) -> tuple[int | None, str | None]:
+    """Process a single attachment and return rating and explanation."""
+    att_ext = (attachment.extension or "").lower()
+    if att_ext not in IMG_FILE_EXTENSIONS:
+        return None, None
+    try:
+        res = await get_meme_rating(attachment.url, username)
+        if not res:
+            return None, None
+        return res.rate, res.explanation
+    except Exception:
+        return None, None
+
+
+async def _process_embed(
+    embed: hikari.Embed, username: str | None
+) -> tuple[int | None, str | None]:
```

Comment:
Is it easier to use `tuple[int, str] | None` or is there a chance that the rating is none and the explanation is not?

Reply:
The way get rater works, having `None` as tuple elements is better.
This can be made as `good first issue` as well.

Reply:
If the meme rater required this format, then perhaps we should change that, too.